src/librustc/hir/check_attr.rs

```diff
@@ -154,7 +154,59 @@ impl<'a, 'tcx> CheckAttrVisitor<'a, 'tcx> {
         let hints: Vec<_> = item.attrs
             .iter()
             .filter(|attr| attr.name() == "repr")
-            .filter_map(|attr| attr.meta_item_list())
+            .filter_map(|attr| {
+                let list = attr.meta_item_list();
+
+                // Emit warnings with `repr` either has a literal assignment (`#[repr = "C"]`) or
```

Comment:
This is now complex enough that I'd like to see it extracted into a helper function, or better yet, pulled out of the flat_map (and then kept in this function as a completely separate step). It really won't kill us to go through the attributes and check for name = "repr" twice, and the way this is written now the reader has long forgotten the `let hints = ` by the time they're through this.